radix-engine-profiling/src/rocks_db_metrics/tests/delete.rs

```diff
@@ -0,0 +1,446 @@
+        prepare_db(
+            &mut substate_db,
+            min_size,
+            max_size,
+            size_step,
+            prepare_db_write_repeats,
+            false,
+        )
+    };
+
+    // reopen database and measure deletion times
+    let mut substate_db = create_store(path.clone());
+
+    println!("Delete test execution");
+    let mut rng = rand::thread_rng();
+
+    // repeat 1 substate commit n-times
+    for i in 0..rounds_count {
+        print!("Round {}/{}\r", i + 1, rounds_count);
+        std::io::stdout().flush().ok();
+
+        // prepare vector with indices of data to draw from
+        let mut size_vector: Vec<usize> = Vec::new();
+        for j in (i..data.len()).step_by(prepare_db_write_repeats) {
+            size_vector.push(j);
+        }
+
+        let mut idx_vector = size_vector.clone();
+
+        for _ in 0..size_vector.len() {
+            assert!(!idx_vector.is_empty());
+            // randomize substate size
+            let idx = rng.gen_range(0..idx_vector.len());
+
+            let mut input_data = DatabaseUpdates::new();
+
+            let mut partition = PartitionUpdates::new();
+            partition.insert(data[idx_vector[idx]].1.clone(), DatabaseUpdate::Delete);
+
+            input_data.insert(data[idx_vector[idx]].0.clone(), partition);
+
+            substate_db.commit(&input_data);
+
+            idx_vector.remove(idx);
+        }
+    }
+
+    discard_spikes(&mut substate_db.commit_delete_metrics.borrow_mut(), 100f32);
+    let rocksdb_output_data =
+        calculate_percent_to_max_points(&mut substate_db.commit_delete_metrics.borrow_mut(), 95f32);
+
+    // prepare data for plot
+    let mut rocksdb_data = Vec::with_capacity(100000);
+    for (k, v) in substate_db.commit_delete_metrics.borrow().iter() {
+        for i in v {
+            rocksdb_data.push((*k as f32, i.as_micros() as f32));
+        }
+    }
+    let original_data = substate_db.commit_delete_metrics.borrow().clone();
+
+    (rocksdb_data, rocksdb_output_data, original_data)
+}
+
+fn test_delete_per_partition_internal<F, S>(
+    n_value: usize,
+    rounds_count: usize,
+    create_store: F,
+) -> (
+    Vec<(f32, f32)>,
+    Vec<(f32, f32)>,
+    BTreeMap<usize, Vec<Duration>>,
+)
+where
+    F: Fn(PathBuf) -> SubstateStoreWithMetrics<S>,
+    S: SubstateDatabase + CommittableSubstateDatabase,
+{
+    // RocksDB part
+    let path = PathBuf::from(r"/tmp/radix-scrypto-db");
+    // clean database
+    std::fs::remove_dir_all(path.clone()).ok();
+
+    let mut rng = rand::thread_rng();
+
+    // prepare database
+    let data_per_round: Vec<Vec<(DbPartitionKey, Vec<DbSortKey>)>> = {
+        let mut substate_db = create_store(path.clone());
+        // fill db with 1_000_000 substates of size 100 bytes under random partitions
+        prepare_db(&mut substate_db, 100, 100, 1, 1000000, false);
+
+        // fill db with additional substates which will be deleted in next step
+        let value_size = 100;
```

Comment:
Can't we just delete entries created by `prepare_db`?

Reply:
We can, but I wanted to preserve basic level of database size, deleting substates generated by `prepare_db` will cause database size to decrease (finally to 0) so it can affect results.